### ISSUE TITLE
I have a question regarding attention dropout

### ISSUE THREAD

**Issue by junrae6454** [Dao-AILab/flash-attention#1064]
Hello,

In the Huggingface Llama's eager implementation, it seems that the dropout index is determined using the following code:
```python
attn_weights = nn.functional.dropout(attn_weights, p=self.attention_dropout, training=self.training)
```
[modeling_llama.py#L336](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L336)

I am curious about the method used to determine the dropout index in flash_attention.

In a deterministic environment, I expected that setting the dropout to 0.0 would result in the same loss for both eager and flash_attn, but this was not the case.

I looked through the Python test code but couldn't figure it out, so I am asking for your help.

**Comment by tridao** [comment]
You can see the dropout testing code here:
https://github.com/Dao-AILab/flash-attention/blob/74b0761ff7efc7b90d4e5aeb529c1b2a09a7458c/tests/test_flash_attn.py#L608

**Comment by junrae6454** [comment]
Thank you!
